doc/use_cli_effectively.md

```diff
@@ -94,44 +95,98 @@ Most update commands in the CLI feature the three generic arguments: `--add`, `-
-     
-    To prevent this, you may use [stop-parsing symbol](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_parsing) `--%` between `az` and arguments like `az --% vm create ...`. This can also solve the above-mentioned special character issue and is recommended whenever any issue happens when invoking a batch script from PowerShell.
-     
-    > The stop-parsing symbol (--%), introduced in PowerShell 3.0, directs PowerShell to refrain from interpreting input as PowerShell commands or expressions.
-    >
-    > When it encounters a stop-parsing symbol, PowerShell treats the remaining characters in the line as a literal.
-     
-    This issue is tracked at https://github.com/PowerShell/PowerShell/issues/1995#issuecomment-539822061
-
-## Generic Resource Commands
-  There may be cases where a service you are interested in does not have CLI command coverage. You can use the `az resource create/show/list/delete/update/invoke-action` commands to work with these resources. A few suggestions here:
-  1. If only `create/update` are involved, consider using `az group deployment create`. Leverage [Azure Quickstart Templates](https://github.com/Azure/azure-quickstart-templates) for working examples.
-  2. Check out the Rest API reference for the request payload, URL and API version. As an example, check out the community's comments on [how to create AppInsights](https://github.com/Azure/azure-cli/issues/5543).
-
-## Working behind a proxy
+9. When running Azure CLI commands in PowerShell, parsing errors will occur when the arguments contain special characters of PowerShell, such as at `@`. You can solve this problem by adding `` ` `` before the special character to escape it, or by enclosing the argument with single or double quotes `'`/`"`. For example, `az group deployment create --parameters @parameters.json` doesn't work in PowerShell because `@` is parsed as a [splatting symbol](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_splatting). To fix this, you may change the argument to `` `@parameters.json`` or `'@parameters.json'`.
+10. When using `--query` with a command, some characters of [JMESPath](https://jmespath.org/specification.html) need to be escaped in the shell:
+    ```sh
+    # Wrong, as the dash needs to be quoted in a JMESPath query
+    $ az version --query azure-cli
+    az version: error: argument --query: invalid jmespath_type value: 'azure-cli'
+
+    # Wrong, as the dash needs to be quoted in a JMESPath query, but quotes are interpreted by Bash
+    $ az version --query "azure-cli"
+    az version: error: argument --query: invalid jmespath_type value: 'azure-cli'
+
+    # Correct
+    $ az version --query '"azure-cli"'
```

Comment:
I suggest either call out that this is corerct on Linux/Mac only or put the correct commands on windows command prompt and powershell as well.

Reply:
Nice one, added as requested. The quoting rules for Command Prompt are also mentioned in item 4. As for PowerShell, the format is weird due to a known bug which I further described in https://github.com/Azure/azure-cli/pull/13419.